### PR TITLE
Fix wso2/product-ei/issues/2076

### PR DESCRIPTION
**Pull request by Shakila**
## Purpose
Fix: https://github.com/wso2/product-ei/issues/2076

## Automation tests
 - Unit tests 
   > Code coverage information
 - Integration tests
   > Details about the test cases and coverage

## Security checks
 - Followed secure coding standards in http://wso2.com/technical-reports/wso2-secure-engineering-guidelines? yes/no
 - Ran FindSecurityBugs plugin and verified report? yes/no
 - Confirmed that this PR doesn't commit any keys, passwords, tokens, usernames, or other secrets? yes/no

### CODE REVIEW

File: modules/core/src/main/java/org/apache/synapse/mediators/transform/PayloadFactoryMediator.java [hunk 1]
```diff
@@ -461,7 +461,7 @@ private void removeIndentations(OMElement element, List<OMText> removables) {
                     // XML escape the result of an expression that produces a literal, if the target format
                     // of the payload is XML.
                     details.setXml(isXML(value));
-                    if (!details.isXml() && XML_TYPE.equals(getType()) && !isJson(value)) {
+                    if (!details.isXml() && XML_TYPE.equals(getType()) && !isJson(value.trim())) {
```
Comment (Manuri) [inline modules/core/src/main/java/org/apache/synapse/mediators/transform/PayloadFactoryMediator.java]: Any possibility of "value" being null?

Comment (Nirothipan) [reply modules/core/src/main/java/org/apache/synapse/mediators/transform/PayloadFactoryMediator.java]: value is null checked in https://github.com/Shakila/wso2-synapse/blob/e9e2e9c7eff4bbed9e20b0dd73c077ad185c6c9a/modules/core/src/main/java/org/apache/synapse/mediators/transform/PayloadFactoryMediator.java#L460

Comment (Shakila) [reply modules/core/src/main/java/org/apache/synapse/mediators/transform/PayloadFactoryMediator.java]: Yes. They have a use-case where they send spaces.

Comment (Manuri) [reply modules/core/src/main/java/org/apache/synapse/mediators/transform/PayloadFactoryMediator.java]: @Nirothipan Cool.. Didn't notice that.

**Pull request merged**